### PR TITLE
Add signUp

### PR DESCRIPTION
**Pull request by guilleSequeiraWolox**
## Summary

Add signUp

## Screenshots

Signup

## Trello Card

https://trello.com/c/UK3kW5Is/49-sign-up-i-layout

### CODE REVIEW

File: react/wbooks/src/App.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    .required('Required'),
+  name: Yup.string()
+    .min(2, 'Too Short!')
+    .max(70, 'Too Long!')
+    .required('Required'),
+  password: Yup.string()
+    .min(2, 'Too Short!')
+    .max(70, 'Too Long!')
+    .required('Required')
+});
```
Comment (mariadelacamara) [inline react/wbooks/src/App.js]: extract string and number in a constants file and import them here.

File: react/wbooks/src/app/screens/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import FormWrapper from '../../components/Formik';
+import CustomField from '../../components/CustomField';
+
+import styles from './styles.module.scss';
+
+const logo = require('../../assets/LogoWolox.png');
+
```
Comment (mariadelacamara) [inline react/wbooks/src/app/screens/Login/index.js]: Please import logo as above
`import logo from '.../`

File: react/wbooks/src/app/screens/Login/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          <div className={styles.formContainer}>
+            <img className={styles.logoImage} src={logo} />
+            <form className={styles.formContent} onSubmit={handleSubmit}>
+              <CustomField type="text" name="name" id="Nombre" {...props} />
+              <CustomField type="text" name="lastName" id="Apellido" {...props} />
+              <CustomField type="email" name="email" id="Email" {...props} />
+              <CustomField type="password" name="password" id="Contraseña" {...props} />
+              <CustomField type="password" name="confirmPassword" id="Confirmar Contraseña" {...props} />
+              <button className={styles.SignupButton} type="submit">
+                Signup
```
Comment (mariadelacamara) [inline react/wbooks/src/app/screens/Login/index.js]: Extract this in a constants file

File: react/wbooks/src/app/screens/Login/index.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+              </button>
+              <button className={styles.loginButton} type="button">
+                Login
```
Comment (mariadelacamara) [inline react/wbooks/src/app/screens/Login/index.js]: same

File: react/wbooks/src/app/screens/Login/index.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          </div>
+        </div>
+      )}
+    </FormWrapper>
+  );
+}
+
+Login.propTypes = {
+  onSubmit: PropTypes.func.isRequired,
+  initialValues: PropTypes.object, // eslint-disable-line react/forbid-prop-types
```
Comment (mariadelacamara) [inline react/wbooks/src/app/screens/Login/index.js]: you can do object shape to validate each object property

for example:
```initialValues: PropTypes.shape({
    email: PropTypes.string,
    id: PropTypes.number
  }),```

https://reactjs.org/docs/typechecking-with-proptypes.html

File: react/wbooks/src/app/screens/Login/index.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import styles from './styles.module.scss';
+
+const logo = require('../../assets/LogoWolox.png');
+
+function Login({ initialValues, validationSchema, onSubmit }) {
+  return (
+    <FormWrapper initialValues={initialValues} validationSchema={validationSchema} handleSubmit={onSubmit}>
+      {({ handleSubmit, ...props }) => (
+        <div className={styles.container}>
+          <div className={styles.formContainer}>
```
Comment (mariadelacamara) [inline react/wbooks/src/app/screens/Login/index.js]: Why don't you use just one div for both styles ?

**Review by mariadelacamara — CHANGES_REQUESTED** [review]

File: react/wbooks/src/app/components/CustomField/index.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      {errors[name] && <p className={styles.inputError}>{errors[name]}</p>}
+    </div>
+  );
+}
+
+CustomField.propTypes = {
+  handleBlur: PropTypes.func.isRequired,
+  handleChange: PropTypes.func.isRequired,
+  handleFocus: PropTypes.func.isRequired,
+  errors: PropTypes.object, // eslint-disable-line react/forbid-prop-types
```
Comment (LucasZibell) [inline react/wbooks/src/app/components/CustomField/index.js]: `PropTypes.shape({})`

File: react/wbooks/src/app/components/CustomField/index.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  id: PropTypes.string,
+  name: PropTypes.string,
+  type: PropTypes.string,
+  values: PropTypes.object // eslint-disable-line react/forbid-prop-types
```
Comment (LucasZibell) [inline react/wbooks/src/app/components/CustomField/index.js]: same

File: react/wbooks/src/app/components/Formik/index.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={handleSubmit}>
+      {props => children({ ...props })}
+    </Formik>
+  );
+}
+
+FormWrapper.propTypes = {
+  handleSubmit: PropTypes.func.isRequired, // eslint-disable-line react/forbid-prop-types
+  initialValues: PropTypes.object, // eslint-disable-line react/forbid-prop-types
```
Comment (LucasZibell) [inline react/wbooks/src/app/components/Formik/index.js]: same

File: react/wbooks/src/app/components/Formik/index.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  validationSchema: PropTypes.object // eslint-disable-line react/forbid-prop-types
```
Comment (LucasZibell) [inline react/wbooks/src/app/components/Formik/index.js]: same

File: react/wbooks/src/app/index.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  lastName: Yup.string()
+    .min(2, i18next.t('Validations:shortLastName'))
+    .max(70, i18next.t('Validations:LongLastName'))
+    .required(i18next.t('Validations:fieldEmpty')),
+  name: Yup.string()
+    .min(2, i18next.t('Validations:shortName'))
+    .max(70, i18next.t('Validations:LongName'))
+    .required(i18next.t('Validations:fieldEmpty')),
+  password: Yup.string().required(i18next.t('Validations:fieldEmpty'))
+});
```
Comment (LucasZibell) [inline react/wbooks/src/app/index.js]: Move this to a validation file

File: react/wbooks/src/app/index.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  password: '',
+  confirmPassword: ''
+};
+
+const handleSubmit = (values, actions) => {
+  setTimeout(() => {
+    alert(JSON.stringify(values, null, 2));
+    actions.setSubmitting(false);
+  }, 1000);
+};
```
Comment (LucasZibell) [inline react/wbooks/src/app/index.js]: This should be created inside the component 🤔
Maybe `useCallback` will help

File: react/wbooks/src/app/index.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+import SignUp from './screens/SignUp';
+import styles from './styles.module.scss';
+
+const initialValues = {
+  name: '',
+  lastName: '',
+  email: '',
+  password: '',
+  confirmPassword: ''
+};
```
Comment (LucasZibell) [inline react/wbooks/src/app/index.js]: Constant file

File: react/wbooks/src/app/index.js [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+    .min(2, i18next.t('Validations:shortName'))
+    .max(70, i18next.t('Validations:LongName'))
+    .required(i18next.t('Validations:fieldEmpty')),
+  password: Yup.string().required(i18next.t('Validations:fieldEmpty'))
+});
+
+function App() {
+  return (
+    <div className={styles.container}>
+      <SignUp initialValues={initialValues} validationSchema={validationSchema} onSubmit={handleSubmit} />
```
Comment (LucasZibell) [inline react/wbooks/src/app/index.js]: I think lot of these props should be on SignUp, i doubt the App component should havee this logic

File: react/wbooks/src/app/screens/SignUp/index.js [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    <FormWrapper initialValues={initialValues} validationSchema={validationSchema} handleSubmit={onSubmit}>
+      {({ handleSubmit, ...props }) => (
+        <div className={styles.formContainer}>
+          <img className={styles.logoImage} src={logoWolox} />
+          <form className={styles.formContent} onSubmit={handleSubmit}>
+            <CustomField type="text" name="name" id="Nombre" {...props} />
+            <CustomField type="text" name="lastName" id="Apellido" {...props} />
+            <CustomField type="email" name="email" id="Email" {...props} />
+            <CustomField type="password" name="password" id="Contraseña" {...props} />
+            <CustomField type="password" name="confirmPassword" id="Confirmar Contraseña" {...props} />
```
Comment (LucasZibell) [inline react/wbooks/src/app/screens/SignUp/index.js]: Why are you adding all this id 🤔  Also... the id should be in english and snake case

**Review by LucasZibell — CHANGES_REQUESTED** [review]

File: react/wbooks/src/app/screens/SignUp/index.js (hunk 15 above)
Comment (guilleSequeiraWolox) [reply react/wbooks/src/app/screens/SignUp/index.js]: I added id becuase if the user press in label for example password, the screen active to write in input password.

File: react/wbooks/src/app/screens/SignUp/index.js [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+  //   name: PropTypes.string,
+  //   password: PropTypes.string
+  // }).isRequired,
+  // validationSchema: PropTypes.shape({
+  //   confirmPassword: PropTypes.func,
+  //   email: PropTypes.func,
+  //   lastName: PropTypes.func,
+  //   name: PropTypes.func,
+  //   password: PropTypes.func
+  // }).isRequired,
```
Comment (LucasZibell) [inline react/wbooks/src/app/screens/SignUp/index.js]: Remove comments

**Review by LucasZibell — CHANGES_REQUESTED** [review]

**Review by LucasZibell — APPROVED** [review]

**Review by mariadelacamara — APPROVED** [review]

**Pull request merged**